### PR TITLE
feat(intersection): generate detection lane using RightOfWay and traffic light arrow

### PR DESCRIPTION
**Pull request by soblin**
## Description

Changed how to set RightOfWay tag on Lanelet2 map at the intersections w/o traffic lights and w/o left/right only traffic lights.
All of the 11 cases listed below are handled.

### straight

image
Screenshot from 2022-10-04 13-44-02

### right

image
Screenshot from 2022-10-04 13-44-11

You can see how the detection lanes change if the traffic light changed to turn right only.

https://user-images.githubusercontent.com/28677420/193737208-e473432b-5914-4619-b74b-50592644edae.mp4

### left

turn left only traffic light was not available so that case is not tested.

image
Screenshot from 2022-10-04 13-44-19

## Related links

Jira link: https://tier4.atlassian.net/browse/T4PB-21423

## Notes for reviewers

I will update lanelet2_extension document as well.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [X] The PR follows the [pull request guidelines].
- [X] The PR has been properly tested.
- [X] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [X] There are no open discussions or they are tracked via tickets.
- [X] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_planner/src/scene_module/intersection/util.cpp [hunk 1]
```diff
@@ -351,72 +360,49 @@ bool getObjectiveLanelets(
...
-  // exclude yield lanelets and ego lanelets from objective_lanelets
-  for (const auto & conflicting_lanelet : conflicting_lanelets) {
-    if (lanelet::utils::contains(yield_lanelets, conflicting_lanelet)) {
-      continue;
-    }
-    if (lanelet::utils::contains(ego_lanelets, conflicting_lanelet)) {
-      continue;
+  // exclude yield lanelets and ego lanelets from detection_lanelets
+  // if assigned lanelet is "straight" with traffic light, detection area is not necessary
+  if (turn_direction == std::string("straight") && has_tl) {
```
Comment (kyoichi-sugahara) [inline planning/behavior_velocity_planner/src/scene_module/intersection/util.cpp:367]: Is this condition necessary?
Explicitly indicate detection are is not necessary under this consition?

**Comment by kyoichi-sugahara** [comment]
without this PR
Screenshot from 2022-10-11 20-43-51


confirmed detection area is shorten when right arrow green signal is on.
image

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM

File: planning/behavior_velocity_planner/src/scene_module/intersection/util.cpp (hunk 1 above)
Comment (tkimura4) [reply planning/behavior_velocity_planner/src/scene_module/intersection/util.cpp:367]: @soblin 
How about this comment?

**Pull request merged**